modules/aruco/src/aruco.cpp

```diff
@@ -618,28 +670,38 @@ static uint8_t _identifyOneCandidate(const Ptr<Dictionary>& dictionary, InputArr
 /**
  * @brief Copy the contents of a corners vector to an OutputArray, settings its size.
  */
-static void _copyVector2Output(vector< vector< Point2f > > &vec, OutputArrayOfArrays out) {
+static void _copyVector2Output(vector< vector< Point2f > > &vec, OutputArrayOfArrays out, const double& scale = 1.0) {
     out.create((int)vec.size(), 1, CV_32FC2);
 
+    vector<Point2f> vec_scaled(4);
     if(out.isMatVector()) {
         for (unsigned int i = 0; i < vec.size(); i++) {
             out.create(4, 1, CV_32FC2, i);
             Mat &m = out.getMatRef(i);
-            Mat(Mat(vec[i]).t()).copyTo(m);
+            for (int p = 0; p < 4; ++p) {
+                vec_scaled[p] = vec[i][p]*scale;
+            }
+            Mat(Mat(vec_scaled).t()).copyTo(m);
```

Comment:
You can initialize Mat directly from `vec` elements than multiply it by constant in one line. Let's discuss the function offline. It looks like it could be done with existing OpenCV primitives easier.